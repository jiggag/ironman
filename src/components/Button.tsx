import React from 'react';
import { Text, TouchableOpacity, StyleSheet } from 'react-native';
import Constant from '../utils/constants';

const Button = ({ onPress, text, buttonStyle, ...rest }) => (
  <TouchableOpacity
    onPress={onPress}
    activeOpacity={0.5}
    style={[styles.button, buttonStyle]}
    {...rest}
  >
    <Text style={styles.buttonText}>
      {text}
    </Text>
  </TouchableOpacity>
);

export default Button;

const styles =  StyleSheet.create({
  button: {
    backgroundColor: Constant.MAIN_COLOR,
    paddingVertical: 14,
    alignItems: 'center',
    borderRadius: 6,
    elevation: 5,
    shadowColor: Constant.SHADOW_COLOR,
    shadowOffset: {
      width: 6,
      height: 4,
    },
    shadowOpacity: 0.2,
    shadowRadius: 4,
  },
  buttonText: {
    color: Constant.WHITE_COLOR,
    fontWeight: '600',
    fontSize: 16,
  },
});
