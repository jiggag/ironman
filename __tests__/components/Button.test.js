import React from 'react';
import { render } from '@testing-library/react-native';
import Button from '../../src/components/Button';

it('Renderer Component: Button', () => {
  const { baseElement } = render(
    <Button onPress={() => {}} text="버튼" buttonStyle={{
      backgroundColor: '#ffffff'
    }} />
  )
  expect(baseElement).toMatchSnapshot();
});