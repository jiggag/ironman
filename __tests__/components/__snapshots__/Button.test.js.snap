// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer Component: Button 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 6,
      "elevation": 5,
      "opacity": 1,
      "paddingVertical": 14,
      "shadowColor": "#333333",
      "shadowOffset": Object {
        "height": 4,
        "width": 6,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 4,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  >
    버튼
  </Text>
</View>
`;
